src/transformStateWithClones.js

```diff
@@ -8,6 +8,27 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const array = [];
+  let stateCopy = {
+    ...state,
+  };
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i]['type'] === 'addProperties') {
+      Object.assign(stateCopy, actions[i]['extraData']);
+      array.push({ ...stateCopy });
+    } else if (actions[i]['type'] === 'removeProperties') {
+      for (const key of actions[i]['keysToRemove']) {
+        delete stateCopy[key];
+      }
+      array.push({ ...stateCopy });
+    } else if (actions[i]['type'] === 'clear') {
+      stateCopy = {};
+      array.push({ ...stateCopy });
+    }
```

Comment:
Let's firstly rewrite it using switch\case

Reply:
Its have no sence, and if/else more comfortable